napari_builtins/_qt/features_table.py

```diff
@@ -0,0 +1,561 @@
+
+        if col == 0:  # index
+            if role in {Qt.ItemDataRole.DisplayRole, Qt.ItemDataRole.EditRole}:
+                return str(self.df.index[row])
+            return None
+
+        value = self.df.iat[row, col - 1]
+        dtype = self.df.dtypes.iat[col - 1]
+
+        # show booleans as respective checkboxes
+        if role == Qt.CheckStateRole and pd.api.types.is_bool_dtype(dtype):
+            return Qt.Checked if value else Qt.Unchecked
+
+        if role in {Qt.ItemDataRole.DisplayRole, Qt.ItemDataRole.EditRole}:
+            # format based on dtype
+            if pd.api.types.is_float_dtype(dtype):
+                return f'{value:.6g}'
+            if pd.api.types.is_integer_dtype(dtype):
+                return f'{value:d}'
+            if pd.api.types.is_datetime64_any_dtype(dtype):
+                return value.strftime('%Y-%m-%d')
+            if pd.api.types.is_bool_dtype(dtype):
+                if role == Qt.ItemDataRole.DisplayRole:
+                    return ''  # do not show True/False text
+                if role == Qt.ItemDataRole.EditRole:
+                    return value  # needed for proper sorting
+            return str(value)
+
+        return None
+
+    def headerData(
+        self,
+        section: int,
+        orientation: Qt.Orientation,
+        role=Qt.ItemDataRole.DisplayRole,
+    ) -> Any:
+        if role != Qt.ItemDataRole.DisplayRole:
+            return None
+        if orientation == Qt.Orientation.Horizontal:
+            # special case for index
+            if section == 0:
+                return self.df.index.name or 'Index'
+            return self.df.columns[section - 1]
+        return self.df.index[section]
+
+    def flags(self, index: QModelIndex) -> Qt.ItemFlags:
+        if not index.isValid():
+            return Qt.ItemFlag.ItemIsEnabled
+
+        col = index.column()
+        # index is read-only
+        if col == 0:
+            return Qt.ItemFlag.ItemIsEnabled | Qt.ItemFlag.ItemIsSelectable
+
+        dtype = self.df.dtypes.iat[col - 1]
+        flags = Qt.ItemFlag.ItemIsSelectable | Qt.ItemFlag.ItemIsEnabled
+
+        if self.editable:
+            # make boolean columns checkable
+            if pd.api.types.is_bool_dtype(dtype):
+                flags |= Qt.ItemFlag.ItemIsUserCheckable
+            else:
+                flags |= Qt.ItemFlag.ItemIsEditable
+
+        return flags
+
+    def setData(
+        self, index: QModelIndex, value: Any, role=Qt.ItemDataRole.EditRole
+    ) -> bool:
+        if not index.isValid():
+            return False
+
+        col = index.column()
+        if col == 0:
+            return False  # index is read-only
+
+        row = index.row()
+        dtype = self.df.dtypes.iat[col - 1]
+
+        # checkboxes
+        if (
+            role == Qt.ItemDataRole.CheckStateRole
+            and pd.api.types.is_bool_dtype(dtype)
+        ):
+            self.df.iat[row, col - 1] = bool(value) == Qt.Checked
```

Comment:
```suggestion
            self.df.iat[row, col - 1] = Qt.CheckState(value) == Qt.Checked
```
So this line is the problem for me, in terms of the checkbox not being able to be checked again.
I suspect you arn't on Qt6 and I am. 
Enum handling has changed. On Qt6 `Qt.Checked` returns `CheckState.Checked`, while `value` is 2.
With my suggestion, when checked, the value is 2, and the `Qt.CheckState(value)` returns `CheckState.Checked`, so everything works.  
I did a pyqt5 install locally and it also works, with `Qt.CheckState(value)` returning 2, same as Qt.Checked.

Reply:
I am actually a bit surprised Qt.Checked works on Qt6. i'd expect it to need Qt.CheckedState.Checked -- my IDE complains about this. Might be something qtpy is handling? it does work, but something to consider for future reference. not sure how pyqt5/pyside2 would handle Qt.CheckedState.Checked

Reply:
Nice catch! Yep I was on qt5. Yeah it's weird that it *partially* worked. Maybe it's a bug in qtpy not properly handling all the cases?

Reply:
No. PyQt6 and PySide6, some time ago, restored old `Qt.MemberName` accesor to keep old code working. We still have a bunch of code that contains this. 

IDE highlights it as it is not present in type stubs as deprecated and expected to be removed. I do not know how to detect this on CI.